crates/bifrost/src/providers/replicated_loglet/replication/checker.rs

```diff
@@ -110,11 +111,12 @@ impl FMajorityResult {
 /// legal write quorum. Nodes are assigned a [`StorageState`] that define their authoritative-ness.
 /// An authoritative node is one that has not lost data and is not empty. Empty in this context
 /// means that it's guaranteed that it has never received data and was never a participant in
-/// previous writes. A node is considered non-authoritative if it's in [`StorageState::DataLoss`]
-/// which means it has lost data and it cannot be considered a reliable participant in some quorum
-/// checks. That said, dataloss doesn't include "corruption". It only means that if a node responds
-/// negatively to a read request that we cannot confidently assume that the data was never written.
-/// Conversely, if the node responds with a record/data, we can use this data safely.
+/// previous writes and will never receive new writes. A node is considered non-authoritative if
```

Comment:
The notion of `empty` seems no longer correct. Right now an empty node is one that is `Disabled`. Should authoritative nodes be those that have not lost data (w/o being not empty)?

Reply:
It's clearer to me now. I think the bit where I got hung up was the statement that `Empty` nodes never received any writes and I was thinking that eventually we will be able to move `ReadWrite` nodes to `Disabled` (== Empty as of today) (e.g. when taking them down permanently) and those nodes might have received some writes. In some other PR, I've seen that you assume that `Disabled` nodes need to have all records trimmed which can be thought of as having never received any writes, I guess.